src/protagonist/API/Features/DeliveryChannels/Validation/DeliveryChannelPolicyDataValidator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Text.Json;
+using DLCS.Core.Collections;
+using DLCS.Model.Assets;
+using IIIF.ImageApi;
+
+namespace API.Features.DeliveryChannels.Validation;
+
+public class DeliveryChannelPolicyDataValidator
+{
+    public bool Validate(string policyDataJson, string channel)
+    {
+        return channel switch
+        {
+            AssetDeliveryChannels.Thumbnails => ValidateThumbnailPolicyData(policyDataJson),
+            AssetDeliveryChannels.Timebased => ValidateTimeBasedPolicyData(policyDataJson),
+            _ => false // This is only for thumbs and iiif-av for now
+        };
+    }
+    
+    private string[]? ParseJsonPolicyData(string policyDataJson)
+    {
+        string[]? policyData;
+        try
+        {
+            policyData = JsonSerializer.Deserialize<string[]>(policyDataJson);
+        }
+        catch(JsonException ex)
+        {
+            return null;
+        }
+
+        return policyData;
+    }
+    
+    private bool ValidateThumbnailPolicyData(string policyDataJson)
+    {
+        var policyData = ParseJsonPolicyData(policyDataJson);
+        
+        if (policyData.IsNullOrEmpty())
+        {
+            return false;
+        }
+        
+        var isInvalid = false;
+        
+        foreach (var sizeValue in policyData)
+        {
+            try { SizeParameter.Parse(sizeValue); }
+            catch
+            {
+                isInvalid = true;
+                break;
+            }
+        }
+        
+        return !isInvalid;
+    }
+
+    private bool ValidateTimeBasedPolicyData(string policyDataJson)
+    {
+        var policyData = ParseJsonPolicyData(policyDataJson);
+        
+        // For now, we only expect a single string value
```

Comment:
I don't think that's true. I could specify that my policy is `[ "audio-mp3-128", "audio-flac-256" ]` if I wanted multiple outputs.